pkg/datapath/linux/config/config.go

```diff
@@ -390,6 +391,17 @@ func (h *HeaderfileWriter) WriteNodeConfig(w io.Writer, cfg *datapath.LocalNodeC
 		cDefinesMap["NATIVE_DEV_MAC_BY_IFINDEX(IFINDEX)"] = macByIfIndexMacro
 		cDefinesMap["IS_L3_DEV(ifindex)"] = isL3DevMacro
 	}
+
+	// Enabling NetfilterCompatibleMode increases BPF instruction memory size, and may
+	// cause issues in kernels have lower memory. Hence enabling, this feature
+	// only in kernels supports higher instruction size.
+	// NetfilterCompatibleMode feature can be enabled on all kernels, once we have support
+	// for adding new tail calls in host device[bpf_host].
+	if option.Config.NetfilterCompatibleMode && probes.NewProbeManager().GetMisc().HaveLargeInsnLimit &&
```

Comment:
BigInsnLimit is already checked in daemon_main, do we need to recheck it here?

Reply:
Technically we don't need another check. But from code readability perspective, it reinforces the underlying fact we need to have  `HaveLargeInsnLimit` to support` NetfilterCompatibleMode` feature.

Reply:
I still don't think it's necessary to duplicate these (if I were to play devil's advocate: why don't we remove check in daemon_main and only check here), but if you decide to leave this as is, then please also add a TODO here so that we don't overlook this check when we lift the condition :)

Reply:
I'm going to leave as it is, Once we support for all kernel. we can get rid of these checks.

Reply:
I'll leave as it is, once we have tail call support we will remove this check. tracking here https://github.com/cilium/cilium/issues/16952